encoding/thrift/inbound.go

```diff
@@ -62,6 +62,46 @@ func (t thriftUnaryHandler) Handle(ctx context.Context, treq *transport.Request,
 		return err
 	}
 
+	reader := bytes.NewReader(buf.Bytes())
+
+	if proto, ok := t.Protocol.(protocol.EnvelopeAgnosticProtocol); ok {
```

Comment:
Can you help me understand how/why this upcasting wouldn't succeed?

Reply:
The cast could only fail if the Thrift inbound were constructed with an alternate Protocol constructor option. There are no other protocols at this time, and we would probably implement EnvelopeAgnosticProtocol if we added a CompactProtocol to complement BinaryProtocol.

So failing this cast is unlikely but possible.